app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationScreen.kt

```diff
@@ -34,35 +40,108 @@ fun ConversationScreen(
+                onCopyMessage = conversationScreenState::copyMessage
+            ),
             content = {
-                ConversationScreenContent(
-                    messages = messages,
-                    onMessageChanged = onMessageChanged,
-                    messageText = conversationViewState.messageText,
-                    onSendButtonClicked = onSendButtonClicked
+                Scaffold(
+                    topBar = {
+                        ConversationScreenTopAppBar(
+                            title = conversationName,
+                            onBackButtonClick = onBackButtonClick,
+                            onDropDownClick = {},
+                            onSearchButtonClick = {},
+                            onVideoButtonClick = {}
+                        )
+                    },
+                    snackbarHost = {
+                        SwipeDismissSnackbarHost(
+                            hostState = conversationScreenState.snackBarHostState,
+                            modifier = Modifier.fillMaxWidth()
+                        )
+                    },
+                    content = {
+                        ConversationScreenContent(
+                            messages = messages,
+                            onMessageChanged = onMessageChanged,
+                            messageText = conversationViewState.messageText,
+                            onSendButtonClicked = onSendButtonClicked,
+                            onShowContextMenu = { message -> conversationScreenState.showEditContextMenu(message) }
+                        )
+                    }
                 )
             }
         )
     }
 }
 
+@Composable
+private fun EditMessageMenuItems(
+    editMessageSource: MessageSource?,
+    onCopyMessage: () -> Unit
+): List<@Composable () -> Unit> {
+    return buildList {
+        add {
+            MenuBottomSheetItem(
+                icon = {
+                    MenuItemIcon(
+                        id = R.drawable.ic_copy,
+                        contentDescription = stringResource(R.string.content_description_block_the_user),
+                    )
+                },
+                title = stringResource(R.string.label_copy),
+                onItemClick = onCopyMessage
+            )
+        }
+        if (editMessageSource == MessageSource.CurrentUser)
+            add {
+                MenuBottomSheetItem(
+                    icon = {
+                        MenuItemIcon(
+                            id = R.drawable.ic_edit,
+                            contentDescription = stringResource(R.string.content_description_edit_the_message)
+                        )
+                    },
+                    title = stringResource(R.string.label_edit),
+                )
+            }
+        add {
+            CompositionLocalProvider(LocalContentColor provides MaterialTheme.colorScheme.error) {
```

Comment:
I think we can use here the `Tint()` extension function to make it prettier

Reply:
will add a TODO for now